### DISCUSSION TITLE
Any example for routes, controller and model?

### DISCUSSION THREAD

**Discussion by marvix** [fastify/fastify#3154] [Q&A] [answered]
I'm stuck with the structure, routes call a controller, and the controller uses a function from a model.
Fastify-mongodb must include the route and the schema, so when I want to get a record I must attach the route and schema, I could not find a way to make db query without the route.

So, is there any example of how to use Fastify-mongodb based on routes, controller and model?

**Comment by marvix** [comment]
Why is hard to make such a structure?
and I could not find any example like this ... I'm new to node world, coming from Laravel, and I read how good is Fastify and it is the replacement for ExpressJs ...  Now, I'm stuck with Fastify, all good, yet not much flexible, so would you please help to achieve this?

![image](https://user-images.githubusercontent.com/3768431/123346433-f2ae0680-d560-11eb-8435-fcce659a8ac9.png)

**Comment by StarpTech** [comment]
You can use https://docs.nestjs.com/ that [supports](https://docs.nestjs.com/techniques/performance#performance-fastify) fastify.

**Comment by RafaelGSS** [comment, marked as answer]
Fastify is an HTTP Framework, we don't enforce any structure or scaffolding, it's up to you as a software engineer to decide it.

We have some repository that you can use as a base if you want:

- https://github.com/fastify/fastify-example-twitter
- https://github.com/fastify/fastify-cli (Generate a base structure to you)
- https://github.com/RafaelGSS/napd (Similar to `fastify-cli` but with a more robust schema)

Also, you can search through Github, there are several repositories to you visualize:

https://github.com/search?q=fastify+structure

**Comment by marvix** [comment]
Thanks a lot, I will check.
